components/selectors/parser.rs

```diff
@@ -313,18 +314,28 @@ impl<Impl: SelectorImpl> SelectorMethods for Component<Impl> {
                         return false;
                     }
                 }
-            },
-            AttrExists(ref selector) |
-            AttrEqual(ref selector, _, _) |
-            AttrIncludes(ref selector, _) |
-            AttrDashMatch(ref selector, _) |
-            AttrPrefixMatch(ref selector, _) |
-            AttrSubstringMatch(ref selector, _) |
-            AttrSuffixMatch(ref selector, _) => {
-                if !visitor.visit_attribute_selector(selector) {
+            }
+
+            AttributeInNoNamespace { ref local_name, ref local_name_lower, .. } |
```

Comment:
This should skip never-matching attribute selectors so the documentation for this is accurate. Either that, or add a never_matching argument and filter that in the current methods.